docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,200 @@
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+
+````
+
+### Cosmos SDK Integration
+
+Cosmos SDK provides a base `Network` struct and a `NewNetwork` constructor that could serve as code example for client application developers and testing tool to be used in conjuction with `simd`.
+
+```
+// NewNetwork returns the default application configuration.
+func NewNetwork(options Options) service.Network {
+	cosmosClient := cosmos.NewClient(fmt.Sprintf("http://%s", options.CosmosEndpoint))
+	tendermintClient := tendermint.NewClient(fmt.Sprintf("http://%s", options.TendermintEndpoint))
+
+	return service.Network{
+		Properties: rosetta.NetworkProperties{
+			Blockchain:          options.Blockchain,
+			Network:             options.Network,
+			SupportedOperations: []string{OperationTransfer},
+		},
+		Adapter: newAdapter(
+			cosmosClient,
+			tendermintClient,
+			properties{
+				Blockchain:  options.Blockchain,
+				Network:     options.Network,
+				AddrPrefix:  options.AddrPrefix,
+				OfflineMode: options.OfflineMode,
+			},
+		),
+	}
+}
+```
+
+#### In-process Execution
+
+Rosetta API service could run within the same execution process of the application; new configuration option and command line flag would be provided:
+
+```
+	if config.Rosetta.Enable {
+     ....
+            get contecxt, flags, etc
+        	...
+            
+            h, err := service.New(
+                service.Options{Port: config.Rosetta.Port},
+                NewRosettaAPINetworkFromConfig(config),
+            )
+            if err != nil {
+            }
+            
+            ...
+            
+            go func() {
+			if err := h.Start(config); err != nil {
+				errCh <- err
+			}
+		    }()
+    }
+
+```
+
+#### Command Line Integration
+
+Client application developers can write a new command to launch a Rosetta API server as a separate process too:
+
+```
+func NewRosettaServiceCmd() *cobra.Command {
+
+    ...
+    cmd := &cobra.Command{
+    	Use:   "vote [proposal-id] [option]",
+        ....
+        
+		RunE: func(cmd *cobra.Command, args []string) error {
+            ....
+            get contecxt, flags, etc
+        	...
+            
+            h, err := service.New(
+                service.Options{Port: config.Rosetta.Port},
+                NewRosettaAPINetworkFromConfig(config),
+            )
+            if err != nil {
+            }
+            
+            ...
+            
+            h.Start()
+        }
+    }
+    ...
+
+}
+```
+
+
+
+### The external Library
+
+Because apart from the Network struct and Adapter implemention there is a lot of code that would be shared across versions, we can provide a repo for the Rosetta Service that will hold only the Service, the Interfaces and the types. This includes all that is not specific to a single version. (Still to decide if we keep an external shared dependency or not.)
```

Comment:
If I recall from our conversation on Friday this was the point you wanted a decision upon? If so, I would suggest moving this up to the decision or the architecture section. I would also be more assertive about this. The proposed solution sounds good to make it the starting point for the discussion.

Reply:
I agree with @SenorMonito here - plus, this should be rephrased IMvHO